src/libsyntax/parse/token.rs

```diff
@@ -277,6 +277,7 @@ impl Token {
             DotDot | DotDotDot | DotDotEq     | // range notation
             Lt | BinOp(Shl)                   | // associated path
             ModSep                            | // global path
+            Lifetime(..)                      | // labeled loop
             Pound                             => true, // expression attributes
             Interpolated(ref nt) => match nt.0 {
                 NtIdent(..) | NtExpr(..) | NtBlock(..) | NtPath(..) => true,
```

Comment:
Could you add `NtLifetime` here as well?

This also needs a test, something like `return 'label: loop {}` will start parsing with `Lifetime(..)` being able to start an expression.